### PR TITLE
Deprecate out= and dtype= parameter in most dataframe methods

### LINKED ISSUE #10777: Deprecate legacy keywords from reductions
Some of those keywords were part of pandas ages ago (before 1.0 was released), so we should rip them out while moving over to dask-expr

dtype keyword:

- sum
- prod
- product
- mean
- var
- std
- cumsum
- cumprod

out keyword:

- sum
- prod
- product
- max
- min
- mean
- var
- std
- skew
- kurtosis
- cumsum
- cumprod
- cummax
- cummin
- clip
- all
- any

### PR DESCRIPTION
**Pull request by crusaderky**
- Closes #10777

NOT deprecated out parameter:
- from_dict
- to_sql
- elementwise ufuncs (e.g. `np.log`)

### CODE REVIEW

**Comment by phofl** [comment]
You could also use the ``_deprecated_kwarg`` decorator

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -6819,6 +6844,7 @@ def handle_out(out, result):
 
         if not isinstance(out, Scalar):
             out._divisions = result.divisions
+        return result
```
Comment (crusaderky) [inline dask/dataframe/core.py:6847]: Changed from None. Align to pandas for numpy ufuncs.

**Comment by crusaderky** [comment]
All failures unrelated. Ready for review.

**Review by phofl — APPROVED** [review]
Thx

**Pull request merged**